src/main/java/com/google/sps/data/UrlRequest.java

```diff
@@ -0,0 +1,74 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+/* Class conatining helper methods to make url requests */
+public final class UrlRequest {
+  /**
+   * Returns the a JSON string with the API response given an URL path and the query parameters.
+   */
+  public static String urlQuery(String basePath, Map<String, String> params) throws IOException {
+    String path = String.format("%s?%s", basePath, getParamsString(params));
+    URL url = new URL(path);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+    connection.setRequestProperty("Accept", "application/json");
+    connection.setRequestProperty("Content-Type", "application/json");
+    int responseCode = connection.getResponseCode(); // TODO: Handle exceptions when reponse code is not 200.
+    InputStreamReader inputStream = new InputStreamReader(connection.getInputStream());
+    StringBuilder contentBuilder = new StringBuilder();
+    try (BufferedReader inputReader = new BufferedReader(inputStream)) {
+      String inputLine;
+      while ((inputLine = inputReader.readLine()) != null) {
+        contentBuilder.append(inputLine);
+      }
+    } finally {
+      connection.disconnect();
+    }
+    return contentBuilder.toString();
+  }
+
+/**
+  * MIT License
+  * 
+  * Copyright (c) 2017 Eugen Paraschiv
+  * 
+  * Permission is hereby granted, free of charge, to any person obtaining a copy
+  * of this software and associated documentation files (the "Software"), to deal
+  * in the Software without restriction, including without limitation the rights
+  * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  * copies of the Software, and to permit persons to whom the Software is
+  * furnished to do so, subject to the following conditions:
+  * 
+  * The above copyright notice and this permission notice shall be included in all
+  * copies or substantial portions of the Software.
+  */
+
+  public static String getParamsString(Map<String, String> params) throws UnsupportedEncodingException {
```

Comment:
This is similar to what I talked about above.  In general you want to avoid throwing errors if you can.  Think about if there is a way you can handle the Exception with having to throw an error.  Maybe you can just drop the parameter, or maybe you can drop and alert the user.  It also ok to return an HTTP error but make sure to return the correct error code.